### PR TITLE
[Tests] Add PHP testing for new roles

### LINKED ISSUE #10690: 🧪 Update phpunit tests to account for new roles
## 🧪 Tests

Update PHPUnit tests to include tests for new roles: Process Operator, Community Manager, and Community Recruiter.

## 🕵️ Details

Tests which definitely need some attention:
- RolePermissionTest.php
- UserPolicyTest.php
- PoolPolicyTest
- PoolCandidatePolicyTest
- UserTest (testRoleAssignmentsScope)
- UserRoleTest
- UpdateUserTeamRolesTest

Furthermore, do a global search for `assertGraphQLErrorMessage('This action is unauthorized.')`, and consider checking the new roles anywhere that is used.

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt tests were properly updated and remains functional/passing.

- [ ] Existing PHPUnit tests which interact with roles also check the new roles now

## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
- [ ] #10621 
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10364
```

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #10690 

## 👋 Introduction

Add a bunch of testing for the three new roles

## 🕵️ Details

Went through and added a bunch of testing, for the most part surfaced no issues. Just one policy file needed an updated method. 

## 🧪 Testing

Mostly just proofreading and recommending changes, or pointing out missed coverage 
Typos possible from all the copy pasting 😛

### CODE REVIEW

File: api/app/GraphQL/Mutations/UpdateUserTeamRoles.php
Comment (vd1992) [inline api/app/GraphQL/Mutations/UpdateUserTeamRoles.php:1]: `UpdateUserTeamRolesTest` was removed in https://github.com/GCTC-NTGC/gc-digital-talent/pull/11353 but this mutation class was missed

**Review by petertgiles — CHANGES_REQUESTED** [review]
Looks great!  Love the new tests.  Let's take a second look at that policy update, though.

File: api/app/Policies/PoolPolicy.php [hunk 1]
```diff
@@ -147,8 +147,18 @@ public function updateDraft(User $user, Pool $pool)
      */
     public function updatePublished(User $user, Pool $pool)
     {
-        return $pool->getStatusAttribute() === PoolStatus::PUBLISHED->name
-            && $user->isAbleTo('update-any-publishedPool');
+        if ($pool->getStatusAttribute() === PoolStatus::DRAFT->name) {
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/app/Policies/PoolPolicy.php]: Aye aye https://github.com/GCTC-NTGC/gc-digital-talent/pull/11409/commits/6b94944269b31fa4aeebefc90ceaf313e093098a

[the remark this answers is not in the document]
Comment (petertgiles) [reply api/app/Policies/PoolPolicy.php]: Thanks, that looks better. :+1:

**Review by petertgiles — APPROVED** [review]

**Pull request merged**